test/isochrone.cc

```diff
@@ -85,30 +104,32 @@ void try_isochrone(loki_worker_t& loki_worker,
     } // iteration required
     else {
       // same geom appx
-      auto actual_geom = rp("/features/" + std::to_string(i) + "/geometry/coordinates" +
-                            (actual_type == "Polygon" ? "/0" : ""))
-                             .Get(response)
-                             ->GetArray();
-      auto expected_geom = rp("/features/" + std::to_string(i) + "/geometry/coordinates" +
-                              (expected_type == "Polygon" ? "/0" : ""))
-                               .Get(expected_response)
-                               ->GetArray();
-
-      std::vector<PointLL> actual, expected;
-      for (size_t j = 0; j < std::max(expected_geom.Size(), actual_geom.Size()); ++j) {
-        if (j < actual_geom.Size()) {
-          auto c = actual_geom[j].GetArray();
-          actual.emplace_back(c[0].GetDouble(), c[1].GetDouble());
-        }
-        if (j < expected_geom.Size()) {
-          auto c = expected_geom[j].GetArray();
-          expected.emplace_back(c[0].GetDouble(), c[1].GetDouble());
+
+      std::string coord_selector = "/features/" + std::to_string(i) + "/geometry/coordinates";
+      if (actual_type != "MultiPolygon") {
+        auto actual_geom =
+            rp(coord_selector + (actual_type == "Polygon" ? "/0" : "")).Get(response)->GetArray();
+        auto expected_geom = rp(coord_selector + (expected_type == "Polygon" ? "/0" : ""))
+                                 .Get(expected_response)
+                                 ->GetArray();
+
+        test_shape_equality(actual_geom, expected_geom);
+
+      } else {
+        uint32_t actual_poly_size = rp(coord_selector).Get(response)->GetArray().Size();
+        uint32_t expected_poly_size = rp(coord_selector).Get(response)->GetArray().Size();
+
+        ASSERT_EQ(actual_poly_size, expected_poly_size);
+        for (uint32_t j = 0; j < actual_poly_size; ++j) {
+
+          // we only compare exterior rings
```

Comment:
I noticed we never compare full geometry, as we leave out the holes. maybe check those as well?

Reply:
Let's do that